frontend/src/app/coworking/room-reservation/reservation-table.service.ts

```diff
@@ -85,19 +96,47 @@ export class ReservationTableService {
     );
   }
 
+  getRoomInformation(): Observable<RoomDetails[]> {
+    return this.http.get<RoomDetails[]>(`/api/coworking/rooms/`);
+  }
+
   draftReservation(
     reservationsMap: Record<string, number[]>,
-    operationStart: Date
-  ): Observable<Reservation> {
-    const selectedRoom: { room: string; availability: number[] } | null =
-      this._findSelectedRoom(reservationsMap);
+    operationStart: Date,
+    users: (PublicProfile | Profile)[]
+  ) {
+    return this.publicToProfile(users).pipe(
+      switchMap((detailedUsers: Profile[]) => {
+        const selectedRoom = this._findSelectedRoom(reservationsMap);
+        if (!selectedRoom) {
+          return throwError(() => new Error('No room selected'));
+        }
 
-    if (!selectedRoom) throw new Error('No room selected');
-    const reservationRequest: ReservationRequest = this._makeReservationRequest(
-      selectedRoom!,
-      operationStart
+        return this.getRoomInformation().pipe(
+          map((rooms) => {
+            const room = rooms.find((room) => room.id === selectedRoom.room);
+            if (!room) {
+              throw new Error('Room not found');
+            }
+            if (users.length < room.capacity) {
+              throw new Error('Not enough people to reserve');
+            }
+            const reservationRequest = this._makeReservationRequest(
+              selectedRoom,
+              operationStart,
+              detailedUsers
+            );
+            return reservationRequest;
+          }),
+          switchMap((reservationRequest) =>
+            this.makeDraftReservation(reservationRequest)
+          ),
+          catchError((err) =>
+            throwError(() => new Error('Not enough people to reserve!'))
+          )
+        );
+      })
```

Comment:
This is a lot of logic to have in a frontend service - it might be better to move this into a backend service connected to an API. There can be errors that are resurfaced from the API if rooms do not exist or if there are not enough users. Thoughts @KrisJordan?

Reply:
DraftReservation has been refactored!